### DISCUSSION TITLE
Adonis 5: How to cast boolean to false and true in a model instead of 0 or 1?

### DISCUSSION THREAD

**Discussion by LeCoupa** [adonisjs/core#2103] [Help] [answered]
For a given model, I have one boolean:

table.boolean('free').defaultTo(false)

Depending on its value, it is saved as 1 or 0 in the MySQL database. When I am returning the model (or multiple instances in a pagination result), I get 0 or 1 for this value.

Is there a way to cast this value to a real boolean (false or true)?

**Comment by LeCoupa** [comment, marked as answer]
Along with the serialized property names, you can also customize the value of a given column by defining a custom serialize method.

```javascript
export default class Post extends BaseModel {
  @column({
    serialize: (value?: Number) => {
      return Boolean(value)
    },
  })
  public published: boolean
}
```

**Comment by Joan1590** [comment]
I solve this issue with this:

`@column({
    consume: (value) => {
      return Boolean(value)
    }
  })
  public webAccess: boolean`

**Comment by filev-svt** [comment]
You can also use the shorter decorator serialization: `@column({ serialize: Boolean })`

However, I find this to be quite an unexpected behavior for booleans in typescript, is there any way to set this serialization globally for all boolean attributes in models?

**Comment by sockenklaus** [comment]
Actually it looks like both methods are necessary to make boolean work as expected:

```js
@column({
    consume: Boolean,
    serialize: Boolean
})
```

- `serialize` affects the serialization of the model into a JSON object which is necessary to act correctly when i.e. sending the object as a response
- `consume` affects the behaviour when using it in the backend, i.e. when using strict equals operators **or** (which w as my problem) when doing bouncer policies like:

```js
public async destroy(user: User) {
    return user.isAdmin
} // this results in an always false authorization when not using consume: Boolean
```

**Reply by pauloimon** [reply]
In my case, just `consume: Boolean` was enough. Thanks!
